cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/rest/CloudControllerRestClientImpl.java

```diff
@@ -884,32 +884,28 @@ public CloudServiceBroker getServiceBroker(String name, boolean required) {
         return serviceInstancesMetadata;
     }
 
-    private <T> List<List<T>> toBatches(List<T> largeList, int maxCharLength) {
+    private <T> List<List<T>> toBatches(Collection<T> largeList, int maxCharLength) {
         if (largeList.isEmpty()) {
             return Collections.emptyList();
         }
+
         List<List<T>> batches = new ArrayList<>();
+        int currentFence = 0, currentIndex = 0;
         batches.add(new ArrayList<>());
-        for (T t : largeList) {
-            List<T> currentBatch = batches.get(batches.size() - 1);
-            if (getCharLength(currentBatch) + t.toString()
-                                               .length() >= maxCharLength) {
+
+        for (T elem : largeList) {
```

Comment:
Rename `elem` to `element`. Abbreviations are evil.

Reply:
Done